File: src/main/java/net/guneyilmaz0/permissionss/event/PlayerGroupChangeEvent.java
package net.guneyilmaz0.permissionss.event;

import cn.nukkit.Player;
import cn.nukkit.command.CommandSender;
import cn.nukkit.event.HandlerList;
import cn.nukkit.event.plugin.PluginEvent;
import lombok.Getter;
import net.guneyilmaz0.permissionss.Main;
import net.guneyilmaz0.permissionss.group.Group;

@Getter
public class PlayerGroupChangeEvent extends PluginEvent {
    private final Player player;
    private final Group newGroup;
    private final Group oldGroup;
    private final CommandSender changer;

    private static final HandlerList handlers = new HandlerList();

    public PlayerGroupChangeEvent(Player player, Group newGroup, Group oldGroup, CommandSender changer) {
        super(Main.getInstance());
        this.player = player;
        this.newGroup = newGroup;
        this.oldGroup = oldGroup;
        this.changer = changer;
    }
}

File: src/main/kotlin/events/PlayerGroupChangeEvent.kt
package net.guneyilmaz0.permissions.events

import cn.nukkit.Player
import cn.nukkit.command.CommandSender
import cn.nukkit.event.HandlerList
import cn.nukkit.event.plugin.PluginEvent
import net.guneyilmaz0.permissions.PermissionsS
import net.guneyilmaz0.permissions.group.Group

@Suppress("unused")
class PlayerGroupChangeEvent(
    val player: Player,
    val newGroup: Group,
    val changer: CommandSender
) : PluginEvent(PermissionsS.instance) {

    companion object {
        val handlers = HandlerList()
    }
}
